tox.ini

```diff
@@ -92,9 +109,10 @@ envlist =
 [testenv]
 deps =
   -c dev-requirements.txt
-  opentelemetry: pytest
-  opentelemetry: pytest-benchmark
-  opentelemetry: flaky
+  lint: -r lint-requirements.txt
```

Comment:
Can't we reuse dev-requirements.txt even if it installs a bit more stuff to avoid creating another place where to keep requirements updated?

Reply:
Actually, we should get rid of `dev-requirements.txt` and have properly named requirements files that represent what virtual environment they correspond to.

Reply:
@xrmx 

Has this been resolved offline? What was the verdict?

Reply:
> I think that is adding a lot more complexity that it is required 😢

That's quite subjective.

Now, being objective, this PR will speed up lint tests and it will make this repo consistent with the contrib repo, [_where we already do this_](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2467).

Also, this PR will add the necessary changes for an upcoming PR (analogous to [this PR](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2658)) that can solve #3814.

Reply:
Discussed this issue in the SIG, agreed to open a new issue to discuss objections to this approach.